cupy/cuda/memory.pyx

```diff
@@ -514,6 +516,40 @@ class PooledMemory(Memory):
     __del__ = free
 
 
+cdef int _compaction_threshold = 512
+
+
+cdef _compaction(SingleDeviceMemoryPool pool, size_t stream_ptr, bint free):
```

Comment:
The term memory compaction reminds me something like https://docs.oracle.com/cd/E13150_01/jrockit_jvm/jrockit/geninfo/diagnos/garbage_collect.html#wp1086917.
Also, we may implement in the future https://github.com/cupy/cupy/issues/536.

Using another term would be better? How about "index_compaction"? or "compact_arena"?